com.unity.netcode.gameobjects/Runtime/NetworkVariable/CollectionSerializationUtility.cs

```diff
@@ -0,0 +1,664 @@
+            writer.WriteByte(0);
+            BytePacker.WriteValuePacked(writer, value.Length);
+            writer.WriteValueSafe(changes);
+            unsafe
+            {
+                var ptr = (T*)value.GetUnsafePtr();
+                var prevPtr = (T*)previousValue.GetUnsafePtr();
+                for (int i = 0; i < value.Length; ++i)
+                {
+                    if (changes.IsSet(i))
+                    {
+                        if (i < previousValue.Length)
+                        {
+                            NetworkVariableSerialization<T>.WriteDelta(writer, ref ptr[i], ref prevPtr[i]);
+                        }
+                        else
+                        {
+                            NetworkVariableSerialization<T>.Write(writer, ref ptr[i]);
+                        }
+                    }
+                }
+            }
+        }
+        public static void ReadNativeArrayDelta<T>(FastBufferReader reader, ref NativeArray<T> value) where T : unmanaged
+        {
+            reader.ReadByteSafe(out byte full);
+            if (full == 1)
+            {
+                value.Dispose();
+                reader.ReadValueSafe(out value, Allocator.Persistent);
+                return;
+            }
+            ByteUnpacker.ReadValuePacked(reader, out int length);
+            var changes = new ResizableBitVector(Allocator.Temp);
+            using var toDispose = changes;
+            {
+                reader.ReadNetworkSerializableInPlace(ref changes);
+
+                var previousLength = value.Length;
+                if (length != value.Length)
+                {
+                    var newArray = new NativeArray<T>(length, Allocator.Persistent);
+                    unsafe
+                    {
+                        UnsafeUtility.MemCpy(newArray.GetUnsafePtr(), value.GetUnsafePtr(), math.min(newArray.Length * sizeof(T), value.Length * sizeof(T)));
+                    }
+                    value.Dispose();
+                    value = newArray;
+                }
+
+                unsafe
+                {
+                    var ptr = (T*)value.GetUnsafePtr();
+                    for (var i = 0; i < value.Length; ++i)
+                    {
+                        if (changes.IsSet(i))
+                        {
+                            if (i < previousLength)
+                            {
+                                NetworkVariableSerialization<T>.ReadDelta(reader, ref ptr[i]);
+                            }
+                            else
+                            {
+                                NetworkVariableSerialization<T>.Read(reader, ref ptr[i]);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        public static void WriteListDelta<T>(FastBufferWriter writer, ref List<T> value, ref List<T> previousValue)
+        {
+            if (value == null || previousValue == null)
```

Comment:
"If this is a full update, then set the full update flag and write the entire list".